test/e2e/specs/editor/various/mentions.spec.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'autocomplete mentions', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.createUser( {
+			username: 'testuser',
+			email: 'testuser@example.com',
+			firstName: 'Jane',
+			lastName: 'Doe',
+			password: 'secret',
+		} );
+	} );
+
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllUsers( 'testuser' );
+	} );
+
+	test( 'should insert mention', async ( { page, editor } ) => {
+		await editor.clickBlockAppender();
+		await page.keyboard.type( 'I am @ad' );
+		await page.locator( '.components-autocomplete__result' ).waitFor();
```

Comment:
Let's wait for the option to be explicitly visible.

```suggestion
		await expect(
			page.locator( 'role=listbox >> role=option[name=/admin/i]' )
		).toBeVisible();
```

(Looks like the autocomplete listbox doesn't have a name yet. Would be good to add it in another issue/PR.)